wp-menu-cart.php

```diff
@@ -439,11 +446,50 @@ public function load_scripts_styles() {
 			'wpmenucart-ajax-assist',
 			'wpmenucart_ajax_assist',
 			array(  
-				'shop_plugin' => isset($this->options['shop_plugin']) ? $this->options['shop_plugin'] : '',
-				'always_display' => isset($this->options['always_display']) ? $this->options['always_display'] : '',
+				'shop_plugin'    => isset( $this->options['shop_plugin'] ) ? $this->options['shop_plugin'] : '',
+				'always_display' => isset( $this->options['always_display'] ) ? $this->options['always_display'] : '',
 			)
 		);
+	}
+
+	public function load_font_in_block_editor() {
+		$suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
+
+		// in order to avoid issues with relative font paths, we parse the CSS file and print it inline
+		ob_start();
+		if ( file_exists( plugin_dir_path( __FILE__ ) . 'assets/css/wpmenucart-font'.$suffix.'.css' ) ) {
+			include( plugin_dir_path( __FILE__ ) . 'assets/css/wpmenucart-font'.$suffix.'.css' ) ;
+		}
+		$font_css = str_replace( '../fonts', plugins_url( '/assets/fonts', __FILE__ ), ob_get_clean() );
+		wp_add_inline_style( 'wp-edit-blocks', $font_css );
+	}
+
+	public function register_cart_navigation_block() {
+		$suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? '' : '.min';
+
+		// The Full Site Editor uses an iframe, the styles must be added using the add_editor_style() function. https://github.com/WordPress/gutenberg/pull/25775#issuecomment-703538383
+		add_editor_style( plugins_url( '/assets/css/wpmenucart-icons'.$suffix.'.css', __FILE__ ) );
```

Comment:
I was using `6.0-alpha` with `Twenty-Twenty-Two`, but testing with `5.9` the result is the same. I might be wrong but I think you can't see the CSS loading like that because this particular CSS is being loaded inside the `iframe` header, not in the admin page itself:

![Captura de ecrã de 2022-03-15 08-58-30](https://user-images.githubusercontent.com/533273/158342293-a6baf36f-1fc3-47b3-a30f-6e12e5f75360.png)

Reply:
That's not loading for me either. I'm using Twenty Twenty Two too, but I have the problem with Bricksy too.
The font is loading inline, but only on the main page, not in the iframe.

Reply:
But for me is loading, just inside the `iframe` header where it suppose to be.

Reply:
I have disabled all plugins except WP Menu Cart & WooCommerce, and it's not loading the styles.
I even tried with the Gutenberg plugin (latest version) but no difference.
Nothing in my browser console, nothing in the PHP logs...

Reply:
Checking xDebug, I can see that this `add_editor_style()` is being called, but I can't see if that happens in the editor too (since it's being called several times, `init` may not be the right hook for this?) 

Diving further into the code, it looks like [WordPress may be doing `wp_remote_get` for these stylesheets](https://github.com/WordPress/WordPress/blob/007dec9ec63775ec0eddc6bd1646313284499d39/wp-includes/block-editor.php#L528-L557)? Seems overkill though...

Could `enqueue_block_editor_assets` be useful? https://developer.wordpress.org/reference/hooks/enqueue_block_editor_assets/

Reply:
That's strange, I even tested in Chrome with incognito and it's working just fine. Can we ask other for testing this just to see their output?

Reply:
I tested that before the last approach, it doesn't work because it enqueues those in the admin page, not inside the iframe. Some remarks I found on this topic:

https://github.com/WordPress/gutenberg/issues/38240#issuecomment-1026273176
https://github.com/WordPress/gutenberg/issues/34763#issuecomment-1030826061
https://github.com/WordPress/gutenberg/issues/34333#issuecomment-914031345

Reply:
More findings:

https://make.wordpress.org/core/2021/06/29/blocks-in-an-iframed-template-editor/#comment-41515
https://github.com/WordPress/gutenberg/issues/33212 (all the discussion about inline iframe CSS)

Some users report that are able to do it using `add_editor_style()`: https://github.com/WordPress/gutenberg/issues/33212#issuecomment-1047968369

But in the end I think this is not yet fully working as expected, they are thinking on passing this to the block settings in this roadmap: https://github.com/WordPress/gutenberg/issues/33346#issue-941461152

Reply:
Any idea how WooCommerce manages to load `woocommerce.css` in the iframe?

Reply:
They have special CSS styles for blocks in `wc-blocks-style.css`, `wc-blocks-editor-style.css` and `wc-blocks-vendors-style.css` which are registered here: 

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/c631ea42feb01f7598540ba68758c7086ff5350e/src/AssetsController.php#L50-L52

Using this function:

https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/trunk/src/AssetsController.php#L218-L227

Reply:
I'm referring to the `woocommerce.css` file that's being loaded in the iframe:
![image](https://user-images.githubusercontent.com/3217773/158559971-b9963f92-99aa-4c00-90c3-d90c340c8bcf.png)

If they can do that, we should be able to do that too!

Reply:
Seems to be done here: https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/4345/files

Reply:
That seems like a workable solution, no?

I still think this will be a temporary hack until WP introduces a more sensible way to load styles directly, but it's fine for now.

Reply:
Yes, I think this will change in the future. I will do that then.

Reply:
Could you test with the last commit?

Reply:
✅ Success!
One thing I noticed, is that this doesn't work when the Gutenberg plugin is active. The URL then looks something like this:
```
themes.php?page=gutenberg-edit-site&postType=page&postId=xx
```
instead of:
```
site-editor.php?postType=wp_template&postId=twentytwentytwo%2F%2Fpage
```
I don't know if there's an easy fix for that.

This appears to be linked to WooCommerce (`wc-blocks-style`), but keep in mind we support EDD too, so we should handle this independently:
https://github.com/wpovernight/wp-menu-cart/blob/4c2de765d9666ff41f3cfb90cbb088f26abecdbb/wp-menu-cart.php#L479